### PR TITLE
feat: Add possibility to pass URLSessionConfiguration + extract fetch logic to separate file

### PR DESCRIPTION
**Pull request by polok**

### CODE REVIEW

File: Sources/FeaturevisorSDK/Instance+Fetch.swift [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import FeaturevisorTypes
```
Comment (fahad19) [inline Sources/FeaturevisorSDK/Instance+Fetch.swift:1]: do we wish to keep `+` in the filename? is it common in Swift?

File: Sources/FeaturevisorSDK/Instance.swift [hunk 2]
```diff
@@ -115,7 +88,8 @@ public struct InstanceOptions {
         onRefresh: Listener? = nil,
         onUpdate: Listener? = nil,
         refreshInterval: Int? = nil,
-        stickyFeatures: StickyFeatures? = nil
+        stickyFeatures: StickyFeatures? = nil,
+        sessionConfiguration: URLSessionConfiguration = .default
```
Comment (fahad19) [inline Sources/FeaturevisorSDK/Instance.swift:92]: we are introducing a new option, but I am glad it's optional and already using a default.

**Review by fahad19 — APPROVED** [review]

File: Sources/FeaturevisorSDK/Instance+Fetch.swift (hunk 1 above)
Comment (maxgallo) [reply Sources/FeaturevisorSDK/Instance+Fetch.swift:1]: To my knowledge is a style thing inherited from Objective-C right? but I'm not the expert here nor I care that much about it

**Review by maxgallo — APPROVED** [review]
LGTM

File: Sources/FeaturevisorSDK/Instance+Fetch.swift (hunk 1 above)
Comment (polok) [reply Sources/FeaturevisorSDK/Instance+Fetch.swift:1]: Yes, it's a common thing for Swift world. In that way, we are pointing out that we have an extension (in this case to `FeaturevisorInstance`) and closing some context of usage here (for our case logic to fetch).

File: Sources/FeaturevisorSDK/Instance.swift (hunk 2 above)
Comment (polok) [reply Sources/FeaturevisorSDK/Instance.swift:92]: We should add the ability to customize the `URLSession` which we are using to fetch the datafile which is happening under Featurevisor initialization. I also changed a little bit a way how we init this object in next PR

**Pull request merged**